source/hands_on/trex.rst

```diff
@@ -0,0 +1,82 @@
+
+Key Equations or Key Graphs
+---------------------------
+The loss for reward function:
+
+.. math::
+
+   \mathcal{L}(\theta)=\mathbb{E}_{\tau_i, \tau_j \sim \Pi}[\xi(\mathcal{P}(\hat J_\theta(\tau_i) < \hat J_\theta(\tau_j)),\tau_i \prec \tau_j)]
+
+The function P is defined as follow:
+
+.. math::
+
+   \mathcal{P}(\hat J_\theta(\tau_i) < \hat J_\theta(\tau_j)) \approx \frac {exp \sum_{s \in \tau_j} \hat r_\theta(s)} {exp \sum_{s \in \tau_i} \hat r_\theta(s) + exp \sum_{s \in \tau_j} \hat r_\theta(s)}
+
+The final loss function is in cross entropy form:
+
+.. math::
+   \mathcal{L}(\theta) = - \sum_{\tau_i \prec \tau_j} log \frac {exp \sum_{s \in \tau_j} \hat r_\theta(s)} {exp \sum_{s \in \tau_i} \hat r_\theta(s) + exp \sum_{s \in \tau_j} \hat r_\theta(s)}
+
+Pseudo-code
+---------------
+.. image:: images/TREX.png
+   :align: center
+   :scale: 110%
+
+Extensions
+----------
+TREX can be combined with the following methods：
+
+    - PPO `Proximal Policy Optimization <https://arxiv.org/pdf/1707.06347.pdf>`_
+
+    - SAC `Soft Actor-Critic <https://arxiv.org/pdf/1801.01290>`_
+
+    Given demonstrations generated from RL algorithms or human knowledge (only **observations** and **rankings** are needed), TREX will infer the reward function of the environment. Then the reward function can be applied to RL algorithms like PPO or SAC to estimate rewards while training.
+
+Implementations
+----------------
+The input of the reward model is observations and its output is the predicted reward value. The default reward model is defined as follows:
+
+.. autoclass:: ding.reward_model.TrexRewardModel
+   :noindex:
+
+
+实验 Benchmark
+------------------
++---------------------+-----------------+-----------------------------------------------------+--------------------------+
+| environment         |best mean reward | evaluation results                                  | config link              |
++=====================+=================+=====================================================+==========================+
+|                     |                 |                                                     |`config_link_l <https://  |
+|                     |                 |                                                     |github.com/opendilab/     |
+|                     |                 |                                                     |DI-engine/tree/main/dizoo/|
+|Lunarlander          |  2M env_step,   |.. image:: images/benchmark/lunarlander_gcl.png      |box2d/lunarlander/config/ |
+|                     |  reward 200     |                                                     |lunarlander_trex_dqn_     |
+|                     |                 |                                                     |config.py>`_              |
++---------------------+-----------------+-----------------------------------------------------+--------------------------+

```

Comment:
adding something that indicates the features of Trex algo. For instance, what is the best suboptimal data you used?